File: algorithms/1 - Two Sum.ts
function twoSum(nums: number[], target: number): number[] {
    const map = nums.reduce((obj, num, i) => {
        obj[num] = i;
        return obj;
    }, {});

    for (let i = 0; i < nums.length; i++)
        if (map[target - nums[i]] > i)
            return [i, map[target - nums[i]]];
};

File: algorithms/ruby/1 - Two Sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  map = nums.map.with_index { |n, idx| [n, idx] }.to_h
  nums.each_with_index do |n, idx|
    if map[target - n].to_i > idx
      return [idx, map[target - n]]
    end
  end
end
